src/button/src/components/Button.test.js

```diff
@@ -0,0 +1,315 @@
+import 'jest-styled-components';
+
+import React from 'react';
+import toJson from 'enzyme-to-json';
+import { mount, shallow } from 'enzyme';
+
+import Button from './Button';
+import DefaultButton from './styled/DefaultButton';
+import PrimaryButton from './styled/PrimaryButton';
+import SecondaryButton from './styled/SecondaryButton';
+import { genRandomString, nope } from '../../../utils/testing/MockUtils';
+
+const DEFAULT_BTN_TXT = 'DEFAULT_BUTTON';
+const PRIMARY_BTN_TXT = 'PRIMARY_BUTTON';
+const SECONDARY_BTN_TXT = 'SECONDARY_BUTTON';
+
+describe('Button', () => {
+
+  describe('mode="default"', () => {
+
+    test('should match snapshot', () => {
+      const wrapper1 = mount(
+        <Button onClick={nope}>
+          { DEFAULT_BTN_TXT }
+        </Button>
+      );
+      const wrapper2 = mount(
+        <Button mode="default" onClick={nope}>
+          { DEFAULT_BTN_TXT }
+        </Button>
+      );
+      expect(toJson(wrapper1)).toMatchSnapshot();
+      expect(toJson(wrapper2)).toMatchSnapshot();
+    });
+
+    test('should render DefaultButton component', () => {
```

Comment:
Can we consolidate some of the tests? Some of them are shallow rendering the same components in identical fashions only to be rendered again with different 'expects'